### PR TITLE
Update dispute respond and timeout logic 

### PR DESCRIPTION
**Pull request by peara**
#100

### CODE REVIEW

**Comment by peara** [comment]
@pinkiebell 
This PR has the new `claimTimeout` logic. I also change `submitProof` to immediately submit result.

I will make another PR to add `maxExecutionDepth`

File: tools/executionPoker/bin.js [hunk 1]
```diff
@@ -8,38 +8,14 @@ const ganache = require('ganache-cli');
...
     // we are done
     process.exit(0);
   }
-
-  async submitProof (disputeId, computationPath) {
-    await super.submitProof(disputeId, computationPath);
-
-    submissionCounter++;
-
-    if (submissionCounter === 2) {
```
Comment (pinkiebell) [inline tools/executionPoker/bin.js]: Also, if we account my first comment, we have to keep this or be smarter to check this 😁.

File: contracts/Verifier.sol [hunk 2]
```diff
@@ -256,26 +258,50 @@ contract Verifier is Ownable, HydratedRuntime {
             dispute.state |= CHALLENGER_VERIFIED;
         }
 
-        if ((dispute.state & SOLVER_VERIFIED) != 0 && (dispute.state & CHALLENGER_VERIFIED) != 0) {
-            // both are verified, solver wins
+        if (dispute.state & SOLVER_VERIFIED != 0) {
```
[the remark this answers is not in the document]
Comment (peara) [reply contracts/Verifier.sol]: From https://github.com/leapdao/solEVM-enforcer/blob/master/contracts/Verifier.sol#L251:L257
```
        if (hash == dispute.solver.right) {
            dispute.state |= SOLVER_VERIFIED;
        }

        if (hash == dispute.challenger.right) {
            dispute.state |= CHALLENGER_VERIFIED;
}
```
If challenger is just challenge for fun (same hash root), both will be verified. Hence, challenger will be slashed.
dispute.state & SOLVER_VERIFIED != 0` means solver is good, isn't it :thinking: 
Or do I miss something here? `

File: contracts/Verifier.sol [hunk 3]
```diff
@@ -256,26 +258,50 @@ contract Verifier is Ownable, HydratedRuntime {
...
+     */
     function claimTimeout(bytes32 disputeId) public {
         Dispute storage dispute = disputes[disputeId];
 
-        require(dispute.timeout > 0, "dispute not exit");
+        require(dispute.timeout > 0, "dispute not exist");
         require(dispute.timeout < block.number, "not timed out yet");
         require(
-            !((dispute.state & SOLVER_VERIFIED) != 0 && (dispute.state & CHALLENGER_VERIFIED) != 0),
+            (dispute.state & SOLVER_VERIFIED) == 0 && (dispute.state & CHALLENGER_VERIFIED) == 0,
```
[the remark this answers is not in the document]
Comment (peara) [reply contracts/Verifier.sol]: I don't understand this one.
I did intent to write `require that both SOLVER_VERIFIED and CHALLENGER_VERIFIED is not true`
Do I use the bit incorrectly?

File: contracts/Verifier.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (pinkiebell) [reply contracts/Verifier.sol]: Yes, right. I had a wrong thinking path 😁

File: contracts/Verifier.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (pinkiebell) [reply contracts/Verifier.sol]: Ditto, wrong thinking path on my side. Nvm 👍

**Pull request merged**